cli/command/service/list.go

```diff
@@ -129,12 +129,12 @@ func printTable(out io.Writer, services []swarm.Service, running, tasksNoShutdow
 			replicas = fmt.Sprintf("%d/%d", running[service.ID], tasksNoShutdown[service.ID])
 		}
 		image := service.Spec.TaskTemplate.ContainerSpec.Image
-		ref, err := distreference.ParseNamed(image)
+		ref, err := reference.ParseNormalizedNamed(image)
 		if err == nil {
-			// update image string for display
-			namedTagged, ok := ref.(distreference.NamedTagged)
+			// update image string for display, (strips any digest)
+			namedTagged, ok := ref.(reference.NamedTagged)
 			if ok {
-				image = namedTagged.Name() + ":" + namedTagged.Tag()
+				image = reference.FamiliarName(namedTagged) + ":" + namedTagged.Tag()
```

Comment:
It would have to use `reference.FamiliarName(reference.WithTag(reference.TrimNamed(namedTagged), namedTagged.Tag()))`

I think the string manipulation approach is probably be fine, since this is being formatted for display rather than any internal use.